packages/peregrine/lib/talons/CreateAccount/useCreateAccount.js

```diff
@@ -67,7 +74,15 @@ export const useCreateAccount = props => {
         async formValues => {
             setIsSubmitting(true);
             try {
-                // Try to create an account with the mutation.
+                // Get source cart id (guest cart id).
+                const sourceCartId = cartId;
+
+                // Clear all cart/customer data from cache and redux.
+                await clearCartDataFromCache(apolloClient);
+                await clearCustomerDataFromCache(apolloClient);
+                await removeCart();
```

Comment:
We should consider being more graceful in handling errors during these callbacks. If an error happens during this workflow what happens to the user's old cart?

Reply:
Ignore me, testing reviews in WebStorm; hopefully this isn't a comment.

Reply:
> I would want to restore the old cartId if possible.

But by the time we've caught an error we've already removed the guest cart data from redux. We would need a way to provide a local cart id (`sourceCartId` here) to redux. Maybe all we need to do is add the param to the `getCartDetails` action?

Reply:
Could you manually write the cartId back to localStorage then call `getCartDetails`? Not sure if that would actually update the store though.

Reply:
No, `getCartDetails` reads from redux state, not from local storage. The only thing that reads cart id from local storage is `createCart`, and only if there is no cart id in the store.

Reply:
I've addressed this by moving the stale data clearing to after the sign in or create account invocations. Once they resolve we can assume we should proceed with removal and with fetching/merging the user's cart.